plugin/pkg/shoot/quotavalidator/admission.go

```diff
@@ -392,11 +392,19 @@ func (q *QuotaValidator) getShootResources(shoot garden.Shoot) (corev1.ResourceL
 			return nil, fmt.Errorf("MachineType %s not found in CloudProfile %s", worker.Machine.Type, cloudProfile.Name)
 		}
 
-		// Get the proper VolumeType
-		for _, element := range volumeTypes {
-			if worker.Volume != nil && element.Name == worker.Volume.Type {
-				volumeType = &element
-				break
+		if worker.Volume != nil {
+			if machineType.Storage != nil {
```

Comment:
Should we add a test case for the case, that `machineType.Storage` is specified?

Reply:
Yes, valid point, I implemented two tests. PTAL.